Code/Framework/AzCore/AzCore/Math/IntersectSegment.h

```diff
@@ -32,32 +32,112 @@ namespace AZ
         //! @return The closest point
         Vector3 LineToPointDistance(const Vector3& s1, const Vector3& s2, const Vector3& p, float& u);
 
-        //! Given segment pq and triangle abc (CCW), returns whether segment intersects
-        //! triangle and if so, also returns the barycentric coordinates (u,v,w)
-        //! of the intersection point.
+        //! SegmentTriangleHitTester is an optimized form of "watertight" segment/triangle collisions that is designed
+        //! to compare a single segment against multiple triangles.
+        //! Construction of the class pre-computes information about the segment itself so that each triangle intersection API call
+        //! can perform less setup work.
+        //! Usage:
+        //!   SegmentTriangleHitTester hitTester(rayStart, rayEnd);
+        //!   for (auto& tri : triangles)
+        //!      if (hitTester.IntersectSegmentTriangle(tri.a, tri.b, tri.c, outNormal, outT))
+        //!         hitPoint = rayStart + (rayEnd - rayStart) * outT;
+        class SegmentTriangleHitTester
+        {
+        public:
+            //! Construct a hit tester for segment pq that can be used to compare against multiple triangles.
+            //! @param p Segment start point.
+            //! @param q Segment end point.
+            SegmentTriangleHitTester(const AZ::Vector3& p, const AZ::Vector3& q);
+
+            //! For the given segment pq and a triangle abc (CCW), returns whether the segment intersects the
+            //! triangle and if so, also returns the time of intersection along the segment and the triangle normal.
+            //! @param a Triangle vertex 1.
+            //! @param b Triangle vertex 2.
+            //! @param c Triangle vertex 3.
+            //! @param normal Triangle normal at the intersection point if segment intersects triangle.
+            //! @param t Time of intersection along the segment [0.0 (p), 1.0 (q)] if segment intersects triangle.
+            //! @return true if the segment intersects the triangle otherwise false.
+            bool IntersectSegmentTriangleCCW(
```

Comment:
question: Can these member functions be `const` or do they mutate state in `SegmentTriangleHitTester`?

Reply:
I've switched them. I was a little mixed on the change, because there's a potential for them to mutate state later on down the road, but they don't currently mutate state so it doesn't hurt to make them const now and only switch later if needed.